astro/src/content/quickstarts/quickstart-go-api.mdx

```diff
@@ -0,0 +1,375 @@
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="make-change-function"
+  lang="go"/>
+
+### Add Security
+
+Now you need to implement the `isAuthorized` function used by the handlers. Now, you'll protect your endpoints based on the `roles` encoded in the JWT you receive from FusionAuth. The decoded payload of a JWT for a `teller` might look like this:
+
+```json
+{
+  "aud": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e",
+  "exp": 1689289585,
+  "iat": 1689285985,
+  "iss": "http://localhost:9011",
+  "sub": "00000000-0000-0000-0000-111111111111",
+  "jti": "ebaa4184-2320-47dd-925b-2e18756c635f",
+  "authenticationType": "PASSWORD",
+  "email": "teller@example.com",
+  "email_verified": true,
+  "applicationId": "e9fdb985-9173-4e01-9d73-ac2d60d1dc8e",
+  "roles": [
+    "teller"
+  ],
+  "auth_time": 1689285985,
+  "tid": "d7d09513-a3f5-401c-9685-34ab6c552453"
+}
+```
+
+You need to validate the token and make sure the user has the appropriate role to access the API. You can do so by adding the following code:
+
+<RemoteCode url={frontmatter.coderoot + "/complete-application/main.go"} tags="authorization"
+  lang="go"/>
+
+There is a lot of code in this function. Let's break it down:
+- First you validate some claims. Notice the `checkAudience` that makes sure the token is intended for this application and `checkIss` to make sure the issuer is as expected.
+- `SetPublicKey`, which is a function we will create later, retrieves the public key for the JWT
+- After the key has been set, the `golang-jwt/jwt` will validate the signature and expiration date.
+- Once the token is validated, you will need to take a look at the roles.
+- The code will look to see which page is being called and then create an array for the roles that are valid for that request in `validRoles`.
+     - The containsRole function will be created as well and that creates an array with the intersection of roles that are allowed and roles that are required. If the length of that intersection is greater than 1 then the user has the proper role.
+- Once the roles have been validated, then the endpoint is served.
+
+Now, you will add the code to set the public key. This function will accept the key id from the token and call the authorization server to get the public key with that key id and put it in the proper format to be used.
+
+<Aside type="note">
+In this example, if the key is already set, it will not call it again. In a production environment you could store this key on the server or set it once one initialization. If your api serves several applications, you will need to get the correct public key for each application if they are different.
```

Comment:
```suggestion
In this example, if the key is already set, it will not call it again. In a production environment you could store this key on the server or set it once on initialization. If your api serves several applications, you will need to get the correct public key for each application if they are different.
```